src/parquet/column_writer.h

```diff
@@ -209,6 +220,8 @@ class PARQUET_EXPORT ColumnWriter {
 
   std::vector<CompressedDataPage> data_pages_;
 
+  std::vector<DictionaryPage> saved_dictionary_page_;
```

Comment:
Should the pages all be kept in the same collection?

Reply:
Will do!